bin/test.sh

```diff
@@ -61,15 +64,24 @@ BOOKINFO_DEPLOYMENTS="details-v1 productpage-v1 ratings-v1 reviews-v1 reviews-v2
 if [ "$SKIP_SETUP" -ne 1 ]; then
     kubectl create ns bookinfo || true
     # We use the first namespace with sidecar injection enabled to determine the control plane's namespace.
-    ISTIO_CONTROL=$(kubectl get namespaces -o=jsonpath='{$.items[:1].metadata.labels.istio-env}' -l istio-env)
+    # Fails in KIND.
+    if [ -z "$ISTIO_CONTROL" ]; then
+        ISTIO_CONTROL=$(kubectl get namespaces -o=jsonpath='{$.items[:1].metadata.labels.istio-env}' -l istio-env)
+    fi
     ISTIO_CONTROL=${ISTIO_CONTROL:-istio-control}
 
-    kubectl -n bookinfo delete -f samples/bookinfo/platform/kube/bookinfo.yaml --ignore-not-found
-    kubectl -n bookinfo delete -f samples/bookinfo/networking/destination-rule-all.yaml --ignore-not-found
-    kubectl -n bookinfo delete -f samples/bookinfo/networking/bookinfo-gateway.yaml --ignore-not-found
+    if [ -z "$SKIP_DELETE" ]; then
```

Comment:
why skip a delete?  Comment may help here.

Reply:
On a fresh environment there is nothing to delete. 

Some CI/CD reuse a cluster - so they need to make sure previous runs are cleared up.